src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/service/VisitService.kt

```diff
@@ -73,53 +83,131 @@ class VisitService(
   }
 
   fun cancelVisit(offenderNo: String, visitId: Long, visitDto: CancelVisitRequest) {
+    val today = LocalDate.now()
+    val visit = visitRepository.findById(visitId).orElseThrow(PrisonerNotFoundException(offenderNo))
+    val visitOutcome = visitOutcomeRepository.findById(VisitOutcomeReason.pk(visitDto.outcome)).orElseThrow()
+
+    val cancVisitStatus = visitStatusRepository.findById(VisitStatus.CANCELLED).orElseThrow()
+    val absEventOutcome = eventOutcomeRepository.findById(EventOutcome.ABS).orElseThrow()
+    val cancEventStatus = eventStatusRepository.findById(EventStatus.CANCELLED).orElseThrow()
+
+    if (visit.visitStatus?.code == "CANC") {
+      throw DataNotFoundException("Visit already cancelled") // or 2xx already done?
```

Comment:
yeah, I recon quietly return a 2xx since else it will go on the DLQ and we try again and again. It is nice if it is idempotent

Reply:
Actually there doesnt seem to be a 2xx code which quite fits. I suppose we could just establish a convention that eg 202 means the cancel was ignored